### PR TITLE
Clear selenium tests code

### PR DESCRIPTION
**Pull request by martinRenou**
- [x] Add some routines for explicit wait
- [x] Add a routine for clicking elements
- [x] Remove the routine for clicking buttons (replaced by click_element(By.ID, "my-button"))
- [x] Use mostly ids instead of css_selector or class_names
- [x] Add some high level routines (login, logout, select_application(index=0), start_application, wait_until_application_started, click_new_entry_button, click_remove_user(index=0), ....)

### CODE REVIEW

**Comment by martinRenou** [comment]
There is still some improvements to do, but I think it's enough for this PR. More routines for the AdminDriverTest will come with more tests.

File: frontend/admin/vue-components/accounting/NewAccountingDialog.vue [hunk 1]
```diff
@@ -48,8 +48,8 @@
           <div class="alert alert-danger" v-if="communicationError">
             <strong>Error:</strong> {{communicationError}}
           </div>
-          <button type="button" class="btn btn-default" @click="close()">Cancel</button>
-          <button class="btn btn-primary" type="submit" :disabled="formstate.$invalid">Submit</button>
+          <button id="modal-cancel-btn" type="button" class="btn btn-default" @click="close()">Cancel</button>
```
Comment (martinRenou) [inline frontend/admin/vue-components/accounting/NewAccountingDialog.vue]: do not put ids in generic components.

File: selenium_tests/AdminDriverTest.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    def click_ok_button(self):
+        self.click_element_located(By.ID, "modal-ok-btn")
+        self.wait_until_modal_closed()
+
+    def click_cancel_button(self):
+        self.click_element_located(By.ID, "modal-cancel-btn")
+        self.wait_until_modal_closed()
+
+    def click_user_policies(self, index=0):
+        self.click_element_located(By.ID, "row-{}-action-0".format(index))
```
Comment (martinRenou) [inline selenium_tests/AdminDriverTest.py]: you need to namespace the id with respect to the table as well.

File: selenium_tests/UserDriverTest.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    def wait_until_application_running(self):
+        self.wait_until_element_present(By.ID, "application")
+
+    def wait_until_application_stopped(self):
+        self.wait_until_text_inside(By.ID, "start-button", "Start")
+
+    def wait_until_application_list_loaded(self):
+        self.wait_until_element_invisible(By.ID, "loading-spinner")
+
+    @contextlib.contextmanager
```
Comment (martinRenou) [inline selenium_tests/UserDriverTest.py]: you don't need the context manager anymore. You can simply use the setup/teardown to login and logout specifically for your test cases.

File: selenium_tests/test_create_new_user.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from selenium_tests.AdminDriverTest import AdminDriverTest
+from selenium.webdriver.common.by import By
+
+
+class TestCreateNewUser(AdminDriverTest):
+    def test_cancel(self):
+        with self.logged_in():
```
Comment (martinRenou) [inline selenium_tests/test_create_new_user.py]: this goes away if you use setup/teardown on the base class.

File: selenium_tests/AdminDriverTest.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+    def click_ok_button(self):
+        self.click_first_button("Ok")
+        self.wait_until_modal_closed()
+
+    def click_cancel_button(self):
+        self.click_first_button("Cancel")
+        self.wait_until_modal_closed()
+
+    def trigger_row_action(self, row, action_name):
```
Comment (stefanoborini) [inline selenium_tests/AdminDriverTest.py]: document better what is meant with row.

File: selenium_tests/AdminDriverTest.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+class AdminDriverTest(RemoteAppDriverTest):
+    def setUp(self):
+        RemoteAppDriverTest.setUp(self)
+        self.login("admin")
+
+    def click_new_entry_button(self):
+        self.click_first_button("Create New Entry")
+
+    def click_submit_button(self):
```
Comment (stefanoborini) [inline selenium_tests/AdminDriverTest.py]: submit of what?

File: selenium_tests/AdminDriverTest.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        self.click_first_button("Submit")
+        self.wait_until_modal_closed()
+
+    def click_ok_button(self):
```
Comment (stefanoborini) [inline selenium_tests/AdminDriverTest.py]: click_dialog_ok_button

File: selenium_tests/RemoteAppDriverTest.py [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+        with contextlib.closing(sqlite3.connect(permissions_db_path)) as db:
+            cur = db.cursor()
+            cur.execute(
+                ("INSERT INTO moz_perms VALUES (1, '{base_url}', "
+                 "'popup', 1, 0, 0, 1474977124357)").format(
+                    base_url=self.base_url))
+            db.commit()
+
+    def wait_until_presence_of_element_located(self, how, what):
```
Comment (stefanoborini) [inline selenium_tests/RemoteAppDriverTest.py]: All of these need docs.

File: selenium_tests/RemoteAppDriverTest.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+
+    def type_text_in_element_located(self, how, what, text):
+        element = self.wait_until_clickability_of_element_located(how, what)
+        element.clear()
+        element.send_keys(text)
+
+    def wait_until_modal_closed(self):
+        return self.wait.until_not(EC.alert_is_present())
+
+    def login(self, username="test"):
```
Comment (stefanoborini) [inline selenium_tests/RemoteAppDriverTest.py]: same for this one.

File: selenium_tests/UserDriverTest.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    def select_application(self, index=0):
+        self.click_first_element_located(By.ID, "application-entry-{}".format(index))
+
+    def open_application_settings(self):
+        self.click_first_element_located(By.ID, "application-settings")
+
+    def stop_application(self):
+        self.click_first_element_located(By.ID, "stop-button")
+
+    def start_application(self):
```
Comment (stefanoborini) [inline selenium_tests/UserDriverTest.py]: docs

File: selenium_tests/test_create_new_user.py [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+    def test_create_and_remove_user(self):
+        self.click_first_element_located(By.LINK_TEXT, "Users")
+
+        self.click_new_entry_button()
+        self.type_text_in_element_located(By.CSS_SELECTOR, ".modal-body > form > div > input", "mrenou")
+        self.click_submit_button()
+
+        # Click remove button
+        self.trigger_row_action("mrenou", "Remove")
```
Comment (stefanoborini) [inline selenium_tests/test_create_new_user.py]: click_row_action_button

**Pull request merged**